packages/iris-grid/src/Formatter.ts

```diff
@@ -0,0 +1,202 @@
+      ],
+      [
+        TableUtils.dataType.INT,
+        new IntegerColumnFormatter(integerFormatOptions),
+      ],
+    ]);
+
+    // Formats indexed by data type and column name
+    this.columnFormatMap = Formatter.makeColumnFormatMap(columnFormattingRules);
+  }
+
+  defaultColumnFormatter: TableColumnFormatter;
+
+  typeFormatterMap: Map<DataType, TableColumnFormatter>;
+
+  columnFormatMap: Map<DataType, Map<string, TableColumnFormat>>;
+
+  /**
+   * Gets columnFormatMap indexed by name for a given column type, creates new Map entry if necessary
+   * @param columnType column type
+   * @param createIfNecessary create new entry if true
+   * @returns Map of format strings indexed by column name or undefined if it doesn't exist
+   */
+  getColumnFormatMapForType(
+    columnType: string,
+    createIfNecessary = false
+  ): Map<string, TableColumnFormat> | undefined {
+    const dataType = TableUtils.getNormalizedType(columnType);
+    if (dataType === null) {
+      log.warn(`Unknown columnType ${columnType}`);
+      return undefined;
+    }
+
+    if (createIfNecessary && !this.columnFormatMap.has(dataType)) {
+      this.columnFormatMap.set(dataType, new Map());
+    }
+    return this.columnFormatMap.get(dataType);
+  }
+
+  /**
+   * Gets a column format object for a given column type and name
+   * @param columnType column type
+   * @param columnName column name
+   * @returns format object or null for Default
+   */
+  getColumnFormat(
+    columnType: string,
+    columnName: string
+  ): TableColumnFormat | null {
+    const columnFormatMap = this.getColumnFormatMapForType(columnType);
+    return columnFormatMap?.get(columnName) ?? null;
+  }
+
+  getColumnTypeFormatter(columnType: string): TableColumnFormatter {
+    const dataType = TableUtils.getNormalizedType(columnType);
+    let columnTypeFormatter = this.defaultColumnFormatter;
+    if (dataType) {
+      columnTypeFormatter =
+        this.typeFormatterMap.get(dataType) ?? columnTypeFormatter;
+    }
+    return columnTypeFormatter;
+  }
+
+  /**
+   * Gets formatted string for a given value, column type and name
+   * @param value Value to format
+   * @param columnType Column type used to determine the formatting settings
+   * @param columnName Column name used to determine the formatting settings
+   * @param formatOverride Format object passed to the formatter in place of the format defined in columnFormatMap
+   */
+  getFormattedString(
+    value: unknown,
+    columnType: string,
+    columnName = '',
+    formatOverride?: TableColumnFormat
+  ): string {
+    if (value == null) {
+      return '';
+    }
+
+    const formatter = this.getColumnTypeFormatter(columnType);
+    const format =
+      formatOverride || this.getColumnFormat(columnType, columnName);
+
+    return formatter.format(value, format ?? undefined);
+  }
+
+  /**
+   * Gets the timeZone name
+   * @returns The time zone name E.g. America/New_York
+   */
+  get timeZone(): string {
```

Comment:
I'll skip this one for now just b/c the type alias doesn't provide any extra safety, just readability. In the case of the map earlier it's nice since it's a 2-layer map. In this case I'd say not as helpful for readability